R/experiment.R

```diff
@@ -1475,6 +1614,14 @@ Experiment <- R6::R6Class(
       invisible(self)
     },
     remove_method = function(name = NULL, ...) {
+      if (is.null(name)) {
+        private$.vary_across_list[["method"]] <- list()
+      } else {
+        vary_across_sublist <- private$.vary_across_list[["method"]][[name]]
+        if (!is.null(vary_across_sublist)) {
+          self$remove_vary_across(method = name)
+        }
+      }
```

Comment:
Thanks for catching this bug! Can we also substitute `private$.do_call_handler()` for `do_call_handler()` in `Experiment$evaluate()` and `Experiment$fit()`?

Reply:
I'm not sure it's necessary in `Experiment$evaluate()` and `Experiment$visualize()` (I think that's what you meant), unless we decide to save the errors (and associated data) to disk. The reason is because those methods aren't parallelized, and therefore the error will be signaled as soon as it is encountered.

The more I think about it, I don't think we should save to disk even when `save=TRUE` because the user should be able to catch the errors on their own and saving those temporary objects will just muddy things.

Reply:
Oh yes - thanks that's what I meant.

That makes sense. I forgot we are just requiring the user to wrap the call inside `tryCatch` anyways, so yeah let's go with your plan. I also agree we should not save to disk.